mlx/backend/common/compiled_cpu.cpp

```diff
@@ -68,10 +68,14 @@ void* compile(
   std::string source_code = source_builder();
   std::string kernel_file_name;
 
-  // Deal with long kernel names. Maximum length for files on macOS is 255
-  // characters. Clip file name with a little extra room and append a 16
-  // character hash.
+  // Deal with long kernel names. Maximum length for filename on macOS is 255
+  // characters, and on Windows the maximum length for whole path is 260. Clip
+  // file name with a little extra room and append a 16 character hash.
+#ifdef _WIN32
```

Comment:
This totally fine to merge.. but with very short file-names it's not inconceivable to get collisions on file-names (which would be bad). Especially [given this](https://stackoverflow.com/questions/7968674/unexpected-collision-with-stdhash). Some ideas to help with that:

I might change the hash to only hash the part of the file name that is clipped to reduce probabilities of collisions. If the prefix is different there won't be a collision anyway. And if it's the same, the suffix must be different and we want to use the suffix for the hash.

We might also want to do some checks on the source.. but that is more involved and we can leave that for future work if it becomes an issue.

Reply:
It will take quite a while before anyone actually starts using the Windows port seriously, I prefer leaving the problem for future.

Reply:
You can at least make this change or let me edit the PR:

```
    auto file_id =
        std::hash<std::string>{}(kernel_name.substr(max_file_name_length - 16));
```

It's simple enough and much safer.

Reply:
Please edit the PR for me 🙏

Reply:
I tried but I don't think I have permission :\. Usually it's the default to allow .. but it didn't work in your case. Check out the guide [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).

Reply:
It is likely because my fork is in an org account, I'll update this PR later.

Reply:
Thank you 🙏

Reply:
Update!